### DISCUSSION TITLE
Bluetooth?? on RPI Pico

### DISCUSSION THREAD

**Discussion by djaus2** [earlephilhower/arduino-pico#1081] [Q&A] [answered]
Is there support here for Bluetooth Serial (BLE) for this BSP and in particular for the RPI Pico?
 Documentation on that says there is BLE and it shares the antennae with tthe WiFi.
Or is this a future item?

```
The CYW43439 wireless chip is connected via SPI to the RP2040. While the CYW43439 supports both 802.11 wireless and Bluetooth, initially Pico W does not have Bluetooth support. Support may be added later, and will use the same SPI interface. If support is added existing hardware may require a firmware update to support Bluetooth, but there will be no hardware modifications needed.
```
https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html

**Comment by earlephilhower** [comment, marked as answer]
No BT support, sorry.  It has to come from the Raspberry Pi folks or some other dedicated hacker.  On the bright side that gives me more time to make sure WiFi is working. :)

**Reply by djaus2** [reply]
FYI I having been working with a Freenve Starter Kit. I've done a port of the ESP8266 Example in the ```Azure/azure-sdk-for-c-arduino``` repository on GitHub. Changes were minimal and it runs OK on the Pico. That is, the WiFi works fine! Thx. I have submitted a Pull Request for it there.

I did try many times to get the ESP8266 Arduino Shield working with an Arduino Uno, but gave up. This paves the wauy for revisting that.

I will now produce a new repostory focused only upon the Pico that exhibits much of the Azure IoT Hub functionality, including sensors, warnings and actuators.

**Comment by theeal** [comment]
> Is there support here for Bluetooth Serial (BLE) for this BSP and in particular for the RPI Pico? Documentation on that says there is BLE and it shares the antennae with tthe WiFi. Or is this a future item?
> 
> ```
> The CYW43439 wireless chip is connected via SPI to the RP2040. While the CYW43439 supports both 802.11 wireless and Bluetooth, initially Pico W does not have Bluetooth support. Support may be added later, and will use the same SPI interface. If support is added existing hardware may require a firmware update to support Bluetooth, but there will be no hardware modifications needed.
> ```
> 
> https://www.raspberrypi.com/documentation/microcontrollers/raspberry-pi-pico.html

https://www.hackster.io/news/raspberry-pi-pico-w-bluetooth-support-is-just-around-the-corner-for-both-c-c-and-micropython-592c42d1170a
There is bluetooth on way!

**Reply by earlephilhower** [reply]
https://github.com/raspberrypi/pico-sdk/issues/1164 is the issue in the Pico-SDK.  Please be aware it's targeted for `1.5.0` and we don't yet have a `1.4.1` release yet, so while BT is coming I would not get too excited, yet.  :)
